src/App.jsx

```diff
@@ -14,33 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const prepareGoods = (goods, { field, isReverse }) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (field) {
+    preparedGoods.sort((good1, good2) => {
+      switch (field) {
+        case SORT_FIELD_NAME: {
+          return good1.localeCompare(good2);
+        }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_FIELD_LENGTH: {
+          return good1.length - good2.length;
+        }
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReverse ? preparedGoods.reverse() : preparedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState(null);
+  const [reverse, setReverse] = useState(false);
```

Comment:
Boolean variables should start from `is` or `has`
```suggestion
  const [isReverse, setIsReverse] = useState(false);
```